chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1152 @@
+    LOG.debug("Sent magic sequence")
+    ack = _rawrecv(radio, 2)
+    LOG.debug("Received magic sequence response")
+    if len(ack) != 2 or ack[1:2] != COMMAND_ACCEPT:
+        if ack:
+            LOG.error("Received: Len=%i Data=%s"
+                      % (len(ack), util.hexprint(ack)))
+        raise errors.RadioError("Radio refused to enter programming mode")
+
+    radio.pipe.reset_input_buffer()
+    _rawsend(radio, b"\x02")
+    ident = _rawrecv(radio, radio._magic_response_length)
+
+    if not ident.startswith(radio._fingerprint):
+        raise errors.RadioError("Radio returned unknown identification string")
+
+    LOG.info("Radio entered programming mode")
+    LOG.debug("Radio identification: %s" % util.hexprint(ident))
+
+    _rawsend(radio, COMMAND_ACCEPT)
+    ack = _rawrecv(radio, 1)
+    if ack != COMMAND_ACCEPT:
+        if ack:
+            LOG.error("Got %s" % util.hexprint(ack))
+        raise errors.RadioError("Radio refused to enter programming mode")
+
+    return ident
+
+
+def _exit_programming_mode(radio):
+    try:
+        radio.pipe.write(b"\x45")
+    except Exception:
+        raise errors.RadioError("Radio refused to exit programming mode")
+
+
+def _get_memory_address(channel_id):
+    block = (channel_id - 1) // 16           # Determines high byte
+    offset = ((channel_id - 1) % 16) * 0x10  # Determines low byte
+
+    return (block, offset)
+
+
+def _read_block(radio, channel_id):
+    address = bytes(_get_memory_address(channel_id))
+
+    try:
+        # 52=read | 20 = hex(32 bytes) = length of 2 channels
+        # + settings without header
+        cmd = b"\x52" + address + b"\x20"
+        _rawsend(radio, cmd)
+        data = _recv(radio, address, 32)
+
+        if address != b"\x00\x00":
+            _rawsend(radio, COMMAND_ACCEPT)
+            response = _rawrecv(radio, 1)
+            if response != COMMAND_ACCEPT:
+                raise errors.RadioError("Radio refused to read block at %04s"
+                                        % util.hexprint(address))
+
+        return data
+    except Exception:
+        raise errors.RadioError("Failed to read block from radio at %04s"
+                                % util.hexprint(address))
+
+
+def _write_block(radio, channel_id, data):
+    address = bytes(_get_memory_address(channel_id))
+
+    try:
+        cmd = b"\x57" + address + b"\x10" + data
+        _rawsend(radio, cmd)
+
+        response = _rawrecv(radio, 1)
+        if response != COMMAND_ACCEPT:
+            raise errors.RadioError("Radio refused to write block at %04s"
+                                    % util.hexprint(address))
+    except Exception:
+        raise errors.RadioError("Failed to write block to radio at %04s"
+                                % util.hexprint(address))
+
+
+def _encode_frequency(freq_mhz):
+    """
+    Encodes a frequency in MHz into 4 bytes.
+    Converts from:
+        MHz → Byte3 Byte2 . Byte1 Byte0
+    Example:
+        431.350 → 00 50 13 43
```

Comment:
This looks like scaled BCD to me, like many other drivers, and you've got it defined as such in your memory map. Why are you hand-encoding it and using `set_raw` to force it into memory?